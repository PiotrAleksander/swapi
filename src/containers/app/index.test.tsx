import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import App from ".";

describe("AppContainer", () => {
  it("renders without crashing", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  it("should render expected elements", () => {
    const { getByText } = render(<App />);

    expect(getByText("Counter")).toBeInTheDocument();
    expect(getByText("Shuffle")).toBeInTheDocument();
    expect(getByText("PEOPLE")).toBeInTheDocument();
  });
});
