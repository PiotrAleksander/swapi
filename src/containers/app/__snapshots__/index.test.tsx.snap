// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContainer renders without crashing 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <header>
          <img
            alt="logo"
            class="App-logo"
            src="logo.svg"
          />
          <a
            class="App-link"
            href="https://swapi.co/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <h2
              class="MuiTypography-root MuiTypography-body1"
            >
              Learn SWAPI
            </h2>
          </a>
        </header>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <h4
            class="MuiTypography-root MuiTypography-body1"
          >
            Counter
          </h4>
          <h5
            class="MuiTypography-root MuiTypography-body1"
          >
            0 : 0
          </h5>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Shuffle
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiFormControl-root resource-dropdown"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                data-shrink="true"
                id="resource-dropdown-label"
              >
                Resource Type
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline lowercased MuiInputBase-formControl MuiInput-formControl"
              >
                <div
                  aria-haspopup="listbox"
                  aria-labelledby="resource-dropdown-label resource-dropdown-select"
                  class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
                  id="resource-dropdown-select"
                  role="button"
                  tabindex="0"
                >
                  PEOPLE
                </div>
                <input
                  type="hidden"
                  value="0"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSelect-icon"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-center"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
